crates/core/src/contracts/erc20.rs

```diff
@@ -0,0 +1,50 @@
+use std::marker::PhantomData;
+
+use reth_primitives::U256;
+use starknet::core::types::{BlockId, FunctionCall};
+use starknet::providers::Provider;
+use starknet_crypto::FieldElement;
+
+use crate::client::constants::selectors::BALANCE_OF;
+use crate::client::errors::EthApiError;
+use crate::client::helpers::DataDecodingError;
+use crate::models::felt::Felt252Wrapper;
+
+pub struct Erc20Contract<P> {
+    pub address: FieldElement,
+    _phantom: PhantomData<P>,
+}
+
+impl<P: Provider + Send + Sync> Erc20Contract<P> {
+    #[must_use]
+    pub fn new(address: FieldElement) -> Self {
+        Self { address, _phantom: PhantomData }
+    }
+
+    pub async fn balance_of(
+        &self,
+        starknet_provider: &P,
+        starknet_address: &FieldElement,
+        block_id: &BlockId,
+    ) -> Result<U256, EthApiError<P::Error>> {
+        // Prepare the calldata for the bytecode function call
+        let calldata = vec![*starknet_address];
+        let request = FunctionCall { contract_address: self.address, entry_point_selector: BALANCE_OF, calldata };
+
+        // Make the function call to get the account balance
+        let result = starknet_provider.call(request, block_id).await?;
+        if result.len() != 2 {
+            return Err(DataDecodingError::InvalidReturnArrayLength {
+                entrypoint: "balance_of".into(),
+                expected: 2,
+                actual: result.len(),
+            }
+            .into());
+        };
+        let low: Felt252Wrapper = (*result.get(0).unwrap()).into(); // safe unwrap
```

Comment:
almost no difference but if we eventually turn on "no-unwrap" rule in clippy, this might fail,
why not directly access the index using *result[0] ? this'll do the same but without unwrap()

(minor change), not v important